src/Utilities.cpp

```diff
@@ -547,57 +547,80 @@ double getSemiMinorRadius(json isd, csm::WarningList *list) {
-    }
+  else if (distortion.compare("radial") == 0) {
+    return DistortionType::RADIAL;
   }
-  return coefficients;
-}
 
-std::vector<double> getTransverseDistortionY(json isd, csm::WarningList *list) {
-  std::vector<double> coefficients;
-  try {
-    coefficients = isd.at("optical_distortion").at("transverse").at("y").get<std::vector<double>>();
-    coefficients.resize(10, 0.0);
-  }
-  catch (...) {
-    if (list) {
-      list->push_back(
-        csm::Warning(
-          csm::Warning::DATA_NOT_AVAILABLE,
-          "Could not parse the transverse distortion model Y coefficients.",
-          "Utilities::getTransverseDistortionY()"));
-    }
+  if (list) {
+    list->push_back(
+      csm::Warning(
+        csm::Warning::DATA_NOT_AVAILABLE,
+        "Could not parse the distortion model.",
+        "Utilities::getDistortionModel()"));
   }
-  return coefficients;
 }
 
-std::vector<double> getRadialDistortion(json isd, csm::WarningList *list) {
+std::vector<double> getDistortionCoeffs(json isd, csm::WarningList *list) {
   std::vector<double> coefficients;
-  try {
-    coefficients = isd.at("optical_distortion").at("radial").at("coefficients").get<std::vector<double>>();
-  }
-  catch (...) {
-    if (list) {
-      list->push_back(
-        csm::Warning(
-          csm::Warning::DATA_NOT_AVAILABLE,
-          "Could not parse the radial distortion model coefficients.",
-          "Utilities::getRadialDistortion()"));
+
+  DistortionType distortion = getDistortionModel(isd);
+
+  switch (distortion) {
+    case DistortionType::TRANSVERSE: {
+      try {
+        std::vector<double> coefficientsX, coefficientsY;
+
+        coefficientsX = isd.at("optical_distortion").at("transverse").at("x").get<std::vector<double>>();
+        coefficientsX.resize(10, 0.0);
+
+        coefficientsY = isd.at("optical_distortion").at("transverse").at("y").get<std::vector<double>>();
+        coefficientsY.resize(10, 0.0);
+
+        coefficients = coefficientsX;
+
+        coefficients.insert(coefficients.end(), coefficientsY.begin(), coefficientsY.end());
+        return coefficients;
+      }
+      catch (...) {
+        if (list) {
+          list->push_back(
+            csm::Warning(
+              csm::Warning::DATA_NOT_AVAILABLE,
+              "Could not parse a set of transverse distortion model coefficients.",
+              "Utilities::getDistortion()"));
+        }
+      }
+    }
+    break;
+    case DistortionType::RADIAL: {
+      try {
+        coefficients = isd.at("optical_distortion").at("radial").at("coefficients").get<std::vector<double>>();
+
+        return coefficients;
+      }
+      catch (...) {
+        if (list) {
+          list->push_back(
+            csm::Warning(
+              csm::Warning::DATA_NOT_AVAILABLE,
+              "Could not parse the radial distortion model coefficients.",
+              "Utilities::getDistortion()"));
+        }
+      }
+      return coefficients;
     }
+    break;
```

Comment:
This should return a warning if it doesn't find any distortion coefficients for radial or transverse.

This method can also end without returning. It either needs a warning and return after the switch statement, or a warning and return in a default case of the switch statement.